src/gui/editors/PianoRoll.cpp

```diff
@@ -659,6 +660,63 @@ void PianoRoll::clearGhostPattern()
 }
 
 
+void PianoRoll::glueNotes()
+{
+	if (hasValidPattern())
+	{
+		if (getSelectedNotes().empty())
+		{
+			TextFloat::displayMessage( tr( "Glue notes failed" ),
+					tr( "Please select notes to glue first." ),
+					embed::getIconPixmap( "glue", 24, 24 ),
+					3000 );
+			return;
+		}
+
+		// Make undo possible
+		m_pattern->addJournalCheckPoint();
+
+		QList<Note *> noteToRemove;
+
+		NoteVector selectedNotes = getSelectedNotes();
+
+		for (Note * note : selectedNotes)
```

Comment:
I love it! Only would change the `if` to:

```cpp
if ((*s)->key() == (*nx)->key()
	&& (*nx)->pos() >= (*s)->pos()
	&& (*nx)->pos() <= (*s)->pos() + (*s)->length())
{
```

and the `setLength` to:
```cpp
(*s)->setLength((*s)->length() +
	((*nx)->pos() + (*nx)->length()) -
	((*s)->pos() + (*s)->length()));
```

Else it will mess up if you try to glue overlapping notes. (isn't shown in the screenshots.)

Reply:
Ah, I didn't test overlapping notes. But you're welcome to use my loop and modify it.

Reply:
For clarity of variable names:
`s` = start note of merge check
`nx` = next note

Reply:
Thanks, I will add that as comments on next commit.

Reply:
If you're going to comment something along the lines of //s = start, isn't it better to just rename the variables to something readable?

The indented >= check in the if statement is only 30 characters, surely there's room to spare for proper variable names.

Reply:
It is! Will do :+1:

Reply:
Renamed the variables to proper ones. 

Also added sorting on note key, else it could miss some glueing. Example:
1. add a note to any octave.
2. add another note to another octave.
3. add another note head to tail with the note in 1.
4. add another note head to tail with the note in 2.

Now try to glue :-P

Reply:
Good catch on the out of order notes!

Reply:
It is still out of oder, need to look into this. The notes their pos isn't sorted anymore after the keys are sorted.

Reply:
It should been fixed now. :-)